gateway/src/apicast/policy/rate_limit/rate_limit.lua

```diff
@@ -139,6 +143,19 @@ local function build_limiters_and_keys(type, limiters, redis, error_settings, se
   return res_limiters, res_keys
 end
 
+local function build_templates(limiters)
+  for _, limiter in ipairs(limiters) do
+    limiter.template_string = TemplateString.new(
+      limiter.key.name, limiter.key.name_type or default_name_type)
+  end
+end
+
+local function interpolate_key_name(context, limiters)
+  for _, limiter in ipairs(limiters) do
+    limiter.key_name = limiter.template_string:render(context)
```

Comment:
This mutates the `limiter` object which is shared across all requests. 

You can just pass `context` to `build_limiters_and_keys` and use it there without storing it in the shared limiter object.